### PR TITLE
fix: Case where there is both a "resourcetype" and path in the "resou…

### PR DESCRIPTION
**Pull request by spg**
…rce"

Fixes https://github.com/PokaInc/arnparse/issues/3

### CODE REVIEW

File: arnparse/arnparse.py [hunk 1]
```diff
@@ -45,3 +42,19 @@ def arnparse(arn_str):
...
+
+def _parse_resource(resource):
+    first_separator_index = -1
+    for idx, c in enumerate(resource):
+        if c in (':', '/'):
+            first_separator_index = idx
+            break
+
+    if first_separator_index != -1:
+        c = resource[first_separator_index]
```
Comment (laurrentt) [inline arnparse/arnparse.py]: Renaming `c` to `separator` would help understand the logic here.

**Review by laurrentt — CHANGES_REQUESTED** [review]

File: arnparse/arnparse.py (hunk 1 above)
Comment (spg) [reply arnparse/arnparse.py]: `c` is actually for `char`, so I'm not sure `separator` is a good name

**Review by laurrentt — DISMISSED** [review]

**Comment by ebickle** [comment]
Looks good. You can improve performance a bit by removing the call to .split and using slice notation instead. E.g. 
```
        separator = resource[first_separator_index]
        resource_type, resource = resource.split(separator, 1)
```
becomes
```
    resource_type = resource[:first_separator_index]
    resource = resource[first_separator_index+1:]
```

A sanity check / unit test for a single delimiter at the end of the original resource string might not be a bad idea if you don't have one already - I'm writing this up from memory and haven't personally tested the edge cases for the code above.

**Comment by spg** [comment]
@ebickle thanks for the suggestion! Done in 6626331

**Review by laurrentt — APPROVED** [review]
Thanks @ebickle for chipping in 👍 Good job @spg
